src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala

```diff
@@ -58,24 +58,46 @@
       ambientTemperature: Temperature,
       state: ThermalGridState,
   ): ThermalEnergyDemand = {
-    /* First get the energy demand of the houses */
-    val houseDemand = house
-      .zip(state.houseState)
-      .map { case (house, state) =>
-        house.energyDemand(
-          tick,
-          ambientTemperature,
-          state,
-        )
+    /* First get the energy demand of the houses but only if inner temperature is below target temperature */
+
+    val houseDemand =
+      house.zip(state.houseState).headOption match {
+        case Some((thermalHouse, lastHouseState)) =>
+          val updatedState = thermalHouse.determineState(
+            tick,
+            lastHouseState,
+            ambientTemperature,
+            lastHouseState.qDot,
+          )
+          if (
+            updatedState._1.innerTemperature < thermalHouse.targetTemperature
+          ) {
+            house
+              .zip(state.houseState)
+              .map { case (house, state) =>
+                house.energyDemand(
+                  tick,
+                  ambientTemperature,
+                  state,
+                )
```

Comment:
Would it be possible to use the above `thermalHouse` and `updatedState` directly here?

Reply:
Good question but this brought me to another point which is a very good catch! 

I guess, this is not correct. But please double check as well: The part above updates the lastState to the current tick. To determine the energyDemand from that tick on, one needs to use the updatedState, not the (last)state. Or?

Reply:
I think you need the updated state if you want to determine the current and future energy demand. If for example the house was heated up before and you use the state that does not include this information, it could lead to the overheating of the house.

Therefore I think you need the current state that includes the current temperature.

Reply:
I agree. Guess there might be improvement labeling the states to avoid these kind of failures. I changed to the updatedState already.